### PR TITLE
DAOS-7301 test: Test scrubber RAS event messages.

### PR DESCRIPTION
**Pull request by rpadma2**
Skip-unit-tests: true
Skip-nlt: true
Skip-fault-injection-test: true
Test-tag: test_scrubber_ssd_auto_eviction always_passes test_tiers

Summary:
 - Check the scrubber RAS event generation during data corruption.

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by rpadma2** [comment]
Test passed : https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13769/9/artifact/Functional%20Hardware%20Medium/scrubber/target_auto_eviction.py/job.log

File: src/tests/ftest/scrubber/target_auto_eviction.py [hunk 1]
```diff
@@ -32,10 +34,31 @@ def test_scrubber_ssd_auto_eviction(self):
         self.create_pool_cont_with_scrubber(pool_prop=pool_prop, cont_prop=cont_prop)
         self.dmg_cmd.pool_query(self.pool.identifier)
         initial_metrics = self.scrubber.get_scrub_corrupt_metrics()
+        t_start = journalctl_time()
+        t_end = journalctl_time(datetime.now() + timedelta(seconds=60))
         self.run_ior_and_check_scruber_status(pool=self.pool, cont=self.container)
         # Wait for a minute for the scrubber to take action and evict target
         # after corruption threshold reached.
+        self.log.info("Sleeping for 60 seconds")
         time.sleep(60)
```
Comment (daltonbohning) [inline src/tests/ftest/scrubber/target_auto_eviction.py:43]: This 60s could be incorrect if, for example, `run_ior_and_check_scruber_status` takes a long time to run.
I think it's better to use the actual timestamp
```suggestion
        t_start = journalctl_time()
        self.run_ior_and_check_scruber_status(pool=self.pool, cont=self.container)
        # Wait for a minute for the scrubber to take action and evict target
        # after corruption threshold reached.
        self.log.info("Sleeping for 60 seconds")
        time.sleep(60)
        t_end = journalctl_time()
```

File: src/tests/ftest/scrubber/target_auto_eviction.py [hunk 2]
```diff
@@ -32,10 +34,31 @@ def test_scrubber_ssd_auto_eviction(self):
...
+        err = "Error gathering system log events"
+        results = get_host_data(hosts=self.hostlist_servers, command=command, text="journalctl",
+                                error=err)
+        self.log.info(results)
+        str_to_match = "Data corruption detected"
+        occurrence = 0
+        for count, _ in enumerate(self.hostlist_servers):
+            occurrence = results[count]["data"].count(str_to_match)
+            if occurrence > 0:
+                break
```
Comment (daltonbohning) [inline src/tests/ftest/scrubber/target_auto_eviction.py:56]: Minor - We could just directly loop over the results
```suggestion
        for host_result in results:
            occurrence = host_result["data"].count(str_to_match)
            if occurrence > 0:
                break
```

File: src/tests/ftest/scrubber/target_auto_eviction.py [hunk 3]
```diff
@@ -32,10 +34,31 @@ def test_scrubber_ssd_auto_eviction(self):
...
+        if occurrence > 0:
+            self.log.info("Data corrupted occurrence %s", occurrence)
+        else:
+            self.log.info("RAS data corrupted messages missing on system logs")
+            self.fail("------Test Failed------")
```
Comment (daltonbohning) [inline src/tests/ftest/scrubber/target_auto_eviction.py:61]: Please add a descriptive message for triaging in case this test fails.

**Review by daltonbohning — CHANGES_REQUESTED** [review]

File: src/tests/ftest/scrubber/target_auto_eviction.py (hunk 1 above)
Comment (rpadma2) [reply src/tests/ftest/scrubber/target_auto_eviction.py]: Agree. Initial code was like the suggestion. Then I got sucked into the journalctl not finding the matching string when I used the get_journalctl method... After fixing the method to use the get_journalctl_command(), it should fine now. 
Changes done. Thanks.

File: src/tests/ftest/scrubber/target_auto_eviction.py (hunk 2 above)
Comment (rpadma2) [reply src/tests/ftest/scrubber/target_auto_eviction.py]: Done.

File: src/tests/ftest/scrubber/target_auto_eviction.py (hunk 3 above)
Comment (rpadma2) [reply src/tests/ftest/scrubber/target_auto_eviction.py]: Done.

**Review by daltonbohning — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]

**Pull request merged**